### PR TITLE
Add package for numworks epsilon

### PR DESCRIPTION
**Pull request by erikbackman**
###### Motivation for this change
Add the package

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120613` at 49451a7a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>numworks-epsilon</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/science/math/numworks-epsilon/default.nix:36:3:
```
   |
36 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

pkg-config is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/science/math/numworks-epsilon/default.nix:24:3:
```
   |
24 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `coreutils`.
Near pkgs/applications/science/math/numworks-epsilon/default.nix:3:3:
```
  |
3 | , coreutils
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    mkdir -p $out/bin
+    cp -r ./output/release/simulator/linux/* $out/bin/
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = ''
+      Emulator for Epsilon, a High-performance graphing calculator operating system
+    '';
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:59]:
```suggestion
    description = "Emulator for Epsilon, a High-performance graphing calculator operating system";
```

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    make PLATFORM=simulator
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    binDir=./output/release/simulator/linux/
+    mv $binDir/epsilon.bin $binDir/epsilon
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:48]:
```suggestion
    mv output/release/simulator/linux/{epsilon.bin,epsilon}
```

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  doBuild = false;
+
+  buildPhase = ''
+    runHook preBuild
+
+    make PLATFORM=simulator clean
+    make PLATFORM=simulator
+
+    runHook postBuild
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:42]: Please use makeFlags.

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  buildInputs = [
+    libpng
+    xorg.libXext
+    python3
+    imagemagick
+    gcc-arm-embedded
+  ];
+
+  doConfigure = false;
+  doBuild = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:33]:
```suggestion
```

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+}:
+
+with lib; stdenv.mkDerivation rec {
+  pname = "numworks-epsilon";
+  version = "15.3.2";
+
+  src = fetchFromGitHub {
+    owner = "numworks";
+    repo = "epsilon";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:19]:
```suggestion
    rev = version;
```

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+, fetchFromGitHub
+, libpng
+, xorg
+, python3
+, imagemagick
+, gcc-arm-embedded
+, pkg-config
+}:
+
+with lib; stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:12]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    binDir=./output/release/simulator/linux/
+    mv $binDir/epsilon.bin $binDir/epsilon
+
+    mkdir -p $out/bin
+    cp -r ./output/release/simulator/linux/* $out/bin/
+
+    runHook postInstall
+  '';
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:56]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please squash this PR into two commits: one for the package and one for the maintainer.

**Pull request merged**